lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -246,6 +246,12 @@ static bool areOverrideCompatibleSimple(ValueDecl *decl,
   if (decl->getKind() != parentDecl->getKind())
     return false;
 
+  // If the parent decl is unavailable, and the child is not looking to override
+  // something, don't count it as an override.
+  if (parentDecl->getAttrs().isUnavailable(parentDecl->getASTContext()))
+    if (!decl->getAttrs().hasAttribute<OverrideAttr>())
```

Comment:
I should probably clarify the comment here: I'm testing for the absence of `OverrideAttr` because, if `override` is present, we want to continue and diagnose it later.

Reply:
I think the`isUnavailable` includes conditionally unavailable declarations as well. Should this use `isUnconditionallyUnavailable` on the attribute instead?

Reply:
I think I want it to be a bit broader than that, but probably not quite as broad as this, so I'll refine it.